aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIOProperties.java

```diff
@@ -428,6 +428,12 @@ public class S3FileIOProperties implements Serializable {
 
   public static final long S3_RETRY_MAX_WAIT_MS_DEFAULT = 20_000; // 20 seconds
 
+  /** Controls whether to list prefixes as directories for S3 Directory buckets */
+  public static final String S3_DIRECTORY_BUCKET_LIST_PREFIX_AS_DIRECTORY =
+      "s3.directory-bucket.list-prefix-as-directories";
```

Comment:
I guess you mean `s3.directory-bucket.list-prefix-as-directory`? the variable and constant name does not match